clients/go/pkg/audit/interceptor.go

```diff
@@ -148,6 +165,41 @@ func (i *Interceptor) UnaryInterceptor(ctx context.Context, req interface{}, inf
 		}
 	}
 
+	// If JWT Validator hasn't been added, we don't handle justification JWTs, and just log
+	// without the justification.
+	if i.jwtValidator != nil {
+		// JWT Validator added, look for justification info
+		md, ok := grpcmetadata.FromIncomingContext(ctx)
+		if !ok {
+			return "", fmt.Errorf("gRPC metadata in incoming context is missing")
+		}
+		vals := md.Get("justification_token")
+		if len(vals) > 0 {
```

Comment:
Style nit: It would be better to reduce the nesting here:

```go
if len(vals) == 0 {
  return i.handleReturnUnary(...)
}

// existing block no need for `else` nesting.
```